### ISSUE TITLE
UserListSerializer and UserListDeserializer should not be used on the approvedBy field of ApprovalRule

### ISSUE THREAD

**Issue by tyates-indeed** [gitlab4j/gitlab4j-api#589]
**Description of Problem**
`UserListSerializer` and `UserListDeserializer` are being applied to the `approvedBy` field of `ApprovalRule`. This is not correct, because this field is a normal list of users and does not need the special serializer and deserializer. You can see this is true by looking at the example from [GitLab's documentation](https://docs.gitlab.com/ee/api/merge_request_approvals.html#get-the-approval-state-of-merge-requests):
```
{
  "approval_rules_overwritten": true,
  "rules": [
    {
      "id": 1,
      "name": "Ruby",
      "rule_type": "regular",
      "eligible_approvers": [
        {
          "id": 4,
          "name": "John Doe",
          "username": "jdoe",
          "state": "active",
          "avatar_url": "https://www.gravatar.com/avatar/0?s=80&d=identicon",
          "web_url": "http://localhost/jdoe"
        }
      ],
      "approvals_required": 2,
      "users": [
        {
          "id": 4,
          "name": "John Doe",
          "username": "jdoe",
          "state": "active",
          "avatar_url": "https://www.gravatar.com/avatar/0?s=80&d=identicon",
          "web_url": "http://localhost/jdoe"
        }
      ],
      "groups": [],
      "contains_hidden_groups": false,
      "approved_by": [
        {
          "id": 4,
          "name": "John Doe",
          "username": "jdoe",
          "state": "active",
          "avatar_url": "https://www.gravatar.com/avatar/0?s=80&d=identicon",
          "web_url": "http://localhost/jdoe"
        }
      ],
      "source_rule": null,
      "approved": true,
      "overridden": false
    }
  ]
}
```
Note that `approved_by` is the exact same as `users`.

`UserListSerializer` and `UserListDeserializer` cause exceptions when trying to interact with merge request approval state JSON because of this mismatch.

**Suggested Fix**
The serializer and deserializer annotations should be removed from the `approvedBy` field of `ApprovalRule`.

**Comment by javydreamercsw** [comment]
Is this related to the missing users in the MR approves field? If so, which version does or will contain this fix?

**Comment by gmessner** [comment]
This has been released in 4.15.3
